pkg/upstream/cluster/loadbalancer.go

```diff
@@ -624,3 +625,98 @@ func (lb *reqRoundRobinLoadBalancer) IsExistsHosts(metadata api.MetadataMatchCri
 func (lb *reqRoundRobinLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int {
 	return lb.hosts.Size()
 }
+
+type shortestResponseLoadBalancer struct {
+	hosts  types.HostSet
+	rand   *rand.Rand
+	mutex  sync.Mutex
+	choice uint32
+}
+
+func newShortestResponseLoadBalancer(info types.ClusterInfo, hosts types.HostSet) types.LoadBalancer {
```

Comment:
There is a check failure reported by reviewdog, which requires we return struct (return `shortestResponseLoadBalancer` instead of `types.LoadBalancer`), but `RegisterLBType` not accepts this function, so this lint cannot be fixed.